tests/Unit/IO/Test_VolumeData.py

```diff
@@ -71,84 +130,39 @@ def test_observation_id(self):
-            -8.66012413502926e-07, -8.66012413502926e-07,
-            -8.66012413502926e-07, -8.66012413502926e-07,
-            -8.66012413502926e-07, -8.66012413502926e-07
-        ])
-        expected_xcoord_tensor_data = np.array([
-            0.0, 6.28318530717959, 0.0, 6.28318530717959, 0.0,
-            6.28318530717959, 0.0, 6.28318530717959
-        ])
-        expected_ycoord_tensor_data = np.array([
-            0.0, 0.0, 6.28318530717959, 6.28318530717959, 0.0, 0.0,
-            6.28318530717959, 6.28318530717959
-        ])
-        expected_zcoord_tensor_data = np.array([
-            0.0, 0.0, 0.0, 0.0, 6.28318530717959, 6.28318530717959,
-            6.28318530717959, 6.28318530717959
-        ])
-        # Checking whether two numpy arrays are "almost equal" is easy, so
-        # we convert everything to numpy arrays for comparison.
-        Psi_tensor_data = np.asarray(
-            vol_file.get_tensor_component(observation_id=obs_id,
-                                          tensor_component='Psi'))
-        npt.assert_array_almost_equal(Psi_tensor_data,
-                                      expected_Psi_tensor_data)
-        Error_tensor_data = np.asarray(
-            vol_file.get_tensor_component(observation_id=obs_id,
-                                          tensor_component='Error(Psi)'))
-        npt.assert_array_almost_equal(Error_tensor_data,
-                                      expected_Error_tensor_data)
-        xcoord_tensor_data = np.asarray(
-            vol_file.get_tensor_component(
-                observation_id=obs_id,
-                tensor_component='InertialCoordinates_x'))
-        npt.assert_array_almost_equal(xcoord_tensor_data,
-                                      expected_xcoord_tensor_data)
-        ycoord_tensor_data = np.asarray(
-            vol_file.get_tensor_component(
-                observation_id=obs_id,
-                tensor_component='InertialCoordinates_y'))
-        npt.assert_array_almost_equal(ycoord_tensor_data,
-                                      expected_ycoord_tensor_data)
-        zcoord_tensor_data = np.asarray(
-            vol_file.get_tensor_component(
-                observation_id=obs_id,
-                tensor_component='InertialCoordinates_z'))
-        npt.assert_array_almost_equal(zcoord_tensor_data,
-                                      expected_zcoord_tensor_data)
+        expected_tensor_components = [
+            'S', 'x-coord', 'y-coord', 'z-coord', 'T_x', 'T_y', 'T_z'
+        ]
+        self.assertEqual(tensor_components, set(expected_tensor_components))
+        # Check Tensor component data
+        for i in range(7):
+            npt.assert_array_almost_equal(
+                np.asarray(
+                    vol_file.get_tensor_component(
+                        observation_id=obs_id,
+                        tensor_component=expected_tensor_components[i]))[0:8],
+                np.asarray(self.tensor_components_and_coords[i]))
```

Comment:
You don't need to convert to numpy arrays and test almost equality. Testing with `self.assertEqual` should be fine since you're not doing any math with the data.

Reply:
@nilsleiffischer How do we know that H5 precision and the python float precision will agree?

Reply:
If you had trouble with this before then better keep the almost-equality testing. I didn't consider the precision changing between reading and writing the data.